classes/Helper.php

```diff
@@ -0,0 +1,122 @@
+     */
+    public function rendercategoricesandroles(): void {
+        GLOBAL $DB, $OUTPUT;
+        $systemcontext = \context_system::instance();
+        $roles = role_fix_names(get_all_roles(), $systemcontext, ROLENAME_ORIGINAL);
+        //$roles = $DB->get_records('role', null, 'id', 'id, shortname');
+        $categorieslist = \core_course_category::make_categories_list();
+        // Now populate $categorieslistandroles with the information to be rendered using mustach-template.
+        $categorieslistandroles = [];
+        foreach ($categorieslist as $categoryid => $categoryname) {
+            $context = \context_coursecat::instance($categoryid);
+            $params = array('contextid' => $context->id);
+            $roleassignments = $DB->get_records('role_assignments', $params, 'contextid, roleid, userid', 'id, contextid, roleid, userid');
+            $roleassignmentsasarray = [];
+            $oldshortname = "";
+            foreach ($roleassignments as $roleassignment) {
+                $counter = false;
+                if (!in_array($roleassignment->roleid, $this->get_supported_roles())) {
+                    continue;
+                }
+                if ($roles[$roleassignment->roleid]->shortname != $oldshortname) {
+                    $oldshortname = $roles[$roleassignment->roleid]->shortname;
+                    $counter = true;
+                }
+                $roleassignmenteditingurl = new moodle_url('/admin/roles/assign.php',
+                    array('contextid' => $context->id , 'roleid' => $roleassignment->roleid));
+                $userprofilgurl = new moodle_url('/user/profile.php',
+                    array('userid' => $roleassignment->userid));
+                $user = \core_user::get_user($roleassignment->userid,  'username, firstname, lastname');
+                $roleassignmentsasarray[] = [
+                    "counter" => $counter,
+                    "roleid" => $roleassignment->roleid,
+                    'roleassignmenteditingurl' => $roleassignmenteditingurl->out(false),
+                    "shortname" => $roles[$roleassignment->roleid]->shortname,
+                    "localname" => $roles[$roleassignment->roleid]->localname,
+                    "userid" => $roleassignment->userid,
+                    "userprofilgurl" => $userprofilgurl->out(false),
+                    "username" => fullname($user)
+                ];
+            }
+            $url = new moodle_url('/course/index.php', array('categoryid' => $categoryid));
+            $categorieslistandroles[] = [
+                'categoryid' => "$categoryid",
+                'categoryname' => "$categoryname",
+                'url' => $url,
+                'roleassignmentsinthiscategory' => $roleassignmentsasarray
+            ];
+        }
+        $data['categorieslistandroles'] = $categorieslistandroles;
+        $translations['header'] = Misc::translate(['page'], 'report_rolessitemap', 'header.');
```

Comment:
that's not necessary. Use {{str}} helper function in the mustache template itself. Do not pass it in the mustache context. BTW: Misc class should be removed completely, does not provide any helpful functions for this plugin